feat: Create the hidden tab (to host both assets and collectibles)

caybro opened the pull request:
- add new Hidden tab/section to Settings/Wallet/Manage tokens page and storybook
- extract the controller(s) to ManageTokensView.qml to share them across the tabs
- adjust the tests

Fixes #13188

### Affected areas

Settings/Wallet/Manage tokens

### Screenshot of functionality (including design for comparison)

- [x] I've checked the design and this PR matches it

![image](https://github.com/status-im/status-desktop/assets/5377645/705cdf3e-9af5-49ec-a703-3677249e0eaa)

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 4aba6383 | [#1](https://ci.status.im/job/status-desktop/job/prs/job/linux/job/x86_64/job/tests-nim/job/PR-13226/1/) | 2024-01-16 16:31:03 | ~5 min | `tests/nim` | [:page_facing_up:`log`](https://ci.status.im/job/status-desktop/job/prs/job/linux/job/x86_64/job/tests-nim/job/PR-13226/1//consoleText) |

micieslak commented on ui/app/AppLayouts/Wallet/panels/ManageHiddenPanel.qml:
```diff
@@ -0,0 +1,227 @@
+
+        StatusListView {
+            Layout.fillWidth: true
+            Layout.preferredHeight: contentHeight
```
We have one model here but if I understand correctly there is ListView unrolling causing everything instantiated up-front, at least when it's put in external scrollable and height of the control is not limited (the case in storybook).

caybro replied:
Yeah thanks, I'll check it out

micieslak commented on ui/app/AppLayouts/Wallet/controls/ManageTokensGroupDelegate.qml:
```diff
@@ -97,25 +94,25 @@ DropArea {
-
                 Item { Layout.fillWidth: true }
 
+                ManageTokensCommunityTag {
```
![Screenshot from 2024-01-17 11-50-17](https://github.com/status-im/status-desktop/assets/20650004/5de61c47-f69b-4285-a207-9eb1394e66d3)
Not sure what's the limit for community name but the tag width should be limited, e.g. to the half of available width and elided when longer.

micieslak commented on storybook/pages/ManageHiddenPanelPage.qml:
The page produces bunch of warnings at startup:
```
status.models.manageTokens: !!! HIDDEN GROUPS MODEL REBUILT WITH GROUPS: ()
status.models.manageTokens: !!! HIDDEN GROUPS MODEL REBUILT WITH GROUPS: ()
status.models.manageTokens: !!! HIDDEN GROUPS MODEL REBUILT WITH GROUPS: ("fpan", "bbrz")
status.models.manageTokens: !!! HIDDEN GROUPS MODEL REBUILT WITH GROUPS: ("ddls", "sox")
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Wallet/controls/ManageTokensDelegate.qml:45: TypeError: Cannot read property 'width' of null
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Wallet/controls/ManageTokensDelegate.qml:45: TypeError: Cannot read property 'width' of null
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Wallet/controls/ManageTokensDelegate.qml:45: TypeError: Cannot read property 'width' of null
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Wallet/controls/ManageTokensDelegate.qml:45: TypeError: Cannot read property 'width' of null
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Wallet/controls/ManageTokensDelegate.qml:45: TypeError: Cannot read property 'width' of null
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Wallet/controls/ManageTokensDelegate.qml:45: TypeError: Cannot read property 'width' of null
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Wallet/controls/ManageTokensGroupDelegate.qml:37: TypeError: Cannot read property 'width' of null
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Wallet/controls/ManageTokensGroupDelegate.qml:37: TypeError: Cannot read property 'width' of null
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Wallet/controls/ManageTokensDelegate.qml:45: TypeError: Cannot read property 'width' of null
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Wallet/controls/ManageTokensDelegate.qml:45: TypeError: Cannot read property 'width' of null
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Wallet/controls/ManageTokensGroupDelegate.qml:37: TypeError: Cannot read property 'width' of null
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Wallet/controls/ManageTokensGroupDelegate.qml:37: TypeError: Cannot read property 'width' of null
```

caybro replied:
Ah the width of the delegate... right, gotta fix that since it's inside a Loader now

micieslak commented on ui/app/AppLayouts/Wallet/panels/ManageHiddenPanel.qml:
```diff
@@ -0,0 +1,227 @@
+            markerRoleName: "tokenType"
+        }
+
+        readonly property var sfpm: SortFilterProxyModel {
```
Instead of adding SFPM to add `isCollectible` and `isGroup` we could use `tokenType` directly in delegates. It would be more efficient than additional proxy on top.

caybro replied:
We (will) need that SFPM anyway for sorting (most recent first), so I just took the opportunity to reuse it for this purpose too

caybro replied:
And no, we can't do it that way; either I use the `ConcatModel` above (with 4 source models, which would create 4 sections); or I use the same tag but then I can't differentiate between single/grouped items

micieslak commented on ui/app/AppLayouts/Wallet/panels/ManageHiddenPanel.qml:
```diff
@@ -0,0 +1,227 @@
+        readonly property int assetsCount: root.assetsController.hiddenTokensModel.count + root.assetsController.hiddenCommunityTokenGroupsModel.count
+        readonly property int collectiblesCount: root.collectiblesController.hiddenTokensModel.count + root.collectiblesController.hiddenCommunityTokenGroupsModel.count
+
+        readonly property var filteredHiddenAssets: SortFilterProxyModel {
+            sourceModel: root.assetsController.hiddenTokensModel
+            filters: FastExpressionFilter {
+                expression: {
+                    root.assetsController.hiddenCommunityGroups
+                    return !root.assetsController.hiddenCommunityGroups.includes(model.communityId)
+                }
+                expectedRoles: ["communityId"]
+            }
+        }
+
+        readonly property var filteredHiddenCollectibles: SortFilterProxyModel {
+            sourceModel: root.collectiblesController.hiddenTokensModel
+            filters: FastExpressionFilter {
+                expression: {
+                    root.collectiblesController.hiddenCommunityGroups
+                    return !root.collectiblesController.hiddenCommunityGroups.includes(model.communityId)
+                }
+                expectedRoles: ["communityId"]
+            }
+        }
```
I would opt for taking already filtered models in this `ManageHiddenPanel` and therefore avoid dependency on controllers here

micieslak replied:
Not sure if we are thinking about the same thing. What I meant was just moving dependency on controllers one level up and making this component taking exactly what it's visualizing.

micieslak reviewed (commented):
Looks nice in general, some doubts/questions only.

caybro commented:
> Looks nice in general, some doubts/questions only.

Thanks, addressed your feedback. I still think we'll have to review together the overall scrolling situation in the settings panels at the sprint :)

status-im-auto commented:
#### :heavy_check_mark: [status-desktop/prs/linux/x86_64/tests-nim/PR-13226#6](https://ci.status.im/job/status-desktop/job/prs/job/linux/job/x86_64/job/tests-nim/job/PR-13226/6/) :small_blue_diamond: ~5 min 56 sec :small_blue_diamond: 7bb5409e :small_blue_diamond: [:package: tests/nim package](https://ci.status.im/job/status-desktop/job/prs/job/linux/job/x86_64/job/tests-nim/job/PR-13226/6//consoleText)
